alert-combiner-py/src/blockchain_indexer_service.py

```diff
@@ -164,4 +164,27 @@ def get_contracts(address, chain_id, disable_etherscan=False, disable_zettablock
                 logging.warning(f"Error getting contract on zettablock for {address}, {chain_id} {e}")
 
         logging.info(f"get_contracts for {address} on {chain_id}; returning {len(contracts)}.")
-        return contracts
\ No newline at end of file
+        return contracts
+    
+    @staticmethod
+    @RateLimiter(max_calls=1, period=1)
+    def get_etherscan_labels(address) -> set:
+        labels_url = f"https://api-metadata.etherscan.io/v1/api.ashx?module=nametag&action=getaddresstag&address={address}&tag=trusted&apikey={BlockChainIndexer.get_api_key(1)}"
+        labels = set()
+        success = False
+        count = 0
+        while not success:
+            data = requests.get(labels_url)
+            if data.status_code == 200:
+                json_data = json.loads(data.content)
+                success = True
+                if "result" in json_data and len(json_data["result"]) > 0:
+                    labels = json_data["result"][0]["labels"]                        
+                return labels
+            else:
+                logging.warning(f"Error getting labels on etherscan: {data.status_code} {data.content}")
+                count += 1
+                if count > 10:
```

Comment:
probably want to add an error log finding here

Reply:
>  I see we do a lookup per address. For the AD, this is likely fine given the volume, but I wonder whether this will work for the scam detector?

In the Scam Detector PR, I'm caching the `findings` before firing the alert and calling Etherscan per 5 findings cached.

> As per the error log finding, could you assess whether there are other APIs we use where this is missing?

Attack detector doesn't seem to emit error findings. I can add it to all calls the way you did for Scam Detector.

@christian-forta